system-addon/content-src/components/TopSites/TopSiteForm.jsx

```diff
@@ -38,44 +39,25 @@ export class TopSiteForm extends React.PureComponent {
 
     if (this.validateForm()) {
       const site = {url: this.cleanUrl()};
+      const index = this.props.index || -1;
       if (this.state.label !== "") {
         site.label = this.state.label;
       }
 
-      // When provided an index we edit a specific TopSite entry
-      if (this.props.index >= 0) {
-        this.editTopSite(site);
-      } else {
-        this.insertTopSite(site);
-      }
+      this.props.dispatch(ac.SendToMain({
+        type: at.TOP_SITES_PIN,
+        data: {site, index}
+      }));
+      this.props.dispatch(ac.UserEvent({
+        source: TOP_SITES_SOURCE,
+        event: "TOP_SITES_EDIT",
+        action_position: index
```

Comment:
We'll probably want to confirm with @ncloudioj (?) that it's okay to get instead a user event for "TOP_SITES_ADD", it'll be a "TOP_SITES_EDIT" with `-1` index.

Reply:
Yes, you can either of them here, and -1 could be used as a valid "action_position". Please consider adding a comment in the `data_dictionary.md` to explain what action_position -1 means for this event.